src/Task.js

```diff
@@ -0,0 +1,50 @@
+import { isUndefined, defer, clearDefer } from './utils';
+
+/**
+ * A utility class to manage a single defferable task.
+ *
+ * @class Task
+ */
+
+export default class Task {
+  _id;
+
+  /**
+   * Checks if a task is set to execute in the future.
+   *
+   * @returns {boolean} True if a task is set.
+   */
+
+  isSet() {
+    return !isUndefined(this._id);
+  }
+
+  /**
+   * Defers a function to be executed on the next turn of the event loop.
+   * Will not allow a defer if the task is already set to execute.
+   *
+   * @param {Function} Callback to execute.
+   */
+
+  defer(callback) {
+    if (this.isSet()) {
+      throw new Error('Cannot set a new task while another is currently set.');
+    }
+
+    this._id = defer(() => {
+      this.reset();
```

Comment:
Yes, oops 😊
Again I wanted to get the basic implementation out there so some of the simple stuff slipped through.